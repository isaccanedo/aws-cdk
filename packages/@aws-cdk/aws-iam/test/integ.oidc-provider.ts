/// !cdk-integ pragma:ignore-assets
import { App, Stack } from '@aws-cdk/core';
import * as iam from '../lib';

const app = new App();
const stack = new Stack(app, 'oidc-provider-integ-test');

new iam.OpenIdConnectProvider(stack, 'Clients', {
  url: 'https://oidc.eks.us-east-1.amazonaws.com/id/test3',
  clientIds: ['foo', 'bar'],
  thumbprints: [
    'aa00aa1122aa00aa1122aa00aa1122aa00aa1122',
    'aa00aa1122aa00aa1122aa00aa1122aa00aa1111',
  ],
});

new iam.OpenIdConnectProvider(stack, 'NoClients', {
  url: 'https://oidc.eks.us-east-1.amazonaws.com/id/test4',
  thumbprints: [
    'aa00aa1122aa00aa1122aa00aa1122aa00aa1122',
    'aa00aa1122aa00aa1122aa00aa1122aa00aa1111',
  ],
});

app.synth();
