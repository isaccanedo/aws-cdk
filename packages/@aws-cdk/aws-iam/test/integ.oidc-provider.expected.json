{
  "Resources": {
    "Clients67031123": {
      "Type": "AWS::IAM::OIDCProvider",
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomAWSCDKOpenIdConnectProviderCustomResourceProviderHandlerF2C543E0",
            "Arn"
          ]
        },
        "ClientIdList": [
          "foo",
          "bar"
        ],
        "Url": "https://oidc.eks.us-east-1.amazonaws.com/id/test3"
      },
      "UpdateReplacePolicy": "Delete",
      "DeletionPolicy": "Delete"
    },
    "Thumbprints9CD6AB02": {
      "Type": "Custom::AWSCDKOpenIdConnectProvider",
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomAWSCDKOpenIdConnectProviderCustomResourceProviderHandlerF2C543E0",
            "Arn"
          ]
        },
        "ThumbprintList": [
          "aa00aa1122aa00aa1122aa00aa1122aa00aa1122",
          "aa00aa1122aa00aa1122aa00aa1122aa00aa1111"
        ],
        "Url": "https://oidc.eks.us-east-1.amazonaws.com/id/test4"
      },
      "UpdateReplacePolicy": "Delete",
      "DeletionPolicy": "Delete"
    }
  }
}
